### PR TITLE
Open Channel/ React: Message Threading

### PR DESCRIPTION
**Pull request by AAPK-code**
#2

https://user-images.githubusercontent.com/67925765/165722691-2c30cab7-75eb-4932-b9d8-63029dd69735.mov

### CODE REVIEW

File: src/samples/OpenChannelMessageThreading.js [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+import { timestampToTime } from '../utils/messageUtils';
+
+let sb;
+
+const BasicOpenChannelSample = (props) => {
+
+    const [state, updateState] = useState({
+        currentlyJoinedChannel: null,
+        currentlyUpdatingChannel: null,
+        treadsParentsMessage: {},
```
Comment (james-a-rob) [inline src/samples/OpenChannelMessageThreading.js:31]: Treads typo.

Would do a search for this typo as I have seen it in a few places.

File: src/samples/OpenChannelMessageThreading.js [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+      </div>
+      <div>{children}</div>
+    </div>;
+
+  }
+  return <div className="channel"></div>;
+
+}
+
+const Threads = ({ isOpenThread, exitThreads, children, threadsParentsMessage, handleDeleteMessage, updateMessage}) => {
```
Comment (james-a-rob) [inline src/samples/OpenChannelMessageThreading.js:449]: Very small but do we not want to use "Thread" singular here and in this component

File: src/samples/OpenChannelMessageThreading.js [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+
+    const paramsThreadedMessageListParams = new ThreadedMessageListParams({
+      prevResultSize: 10,
+      nextResultSize: 10,
+      isInclusive: true,
+      reverse: false,
+      includeParentMessageInfo: false,
+    })
+
+    const { threadedMessages } = await parentsMessage.getThreadedMessagesByTimestamp(30, paramsThreadedMessageListParams);
```
Comment (james-a-rob) [inline src/samples/OpenChannelMessageThreading.js:285]: Can this go into its own function below. A function which also includes the param construction on the lines above

File: src/samples/OpenChannelMessageThreading.js [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
+
+const Thread = ({ isOpenThread, exitThread, children, threadParentsMessage, handleDeleteMessage, updateMessage }) => {
+  return isOpenThread && (
+    <div className="channel thread">
+      <ChannelHeader>Thread</ChannelHeader>
+      <div>
+        <button className="leave-channel" onClick={() => exitThread()}>Exit Thread</button>
+      </div>
+      <Message
+        displayNone={"display-none"}
```
Comment (james-a-rob) [inline src/samples/OpenChannelMessageThreading.js:457]: Passing around a specific display none prop seem a bit much here. Can we not infere when to display the thread based on a property in the state object. perhaps threadMessages.

**Review by petru-gherghel — APPROVED** [review]

**Pull request merged**